sdk/lusid/tcp/tcp_keep_alive_probes.py

```diff
@@ -38,12 +38,12 @@ def adjust_connection_socket(conn, protocol: str = "https"):
         # TCP Keep Alive Probes for different platforms
         platform = sys.platform
         # TCP Keep Alive Probes for Linux
-        if platform == 'linux' and hasattr(socket, "TCP_KEEPIDLE") and hasattr(socket, "TCP_KEEPINTVL") and hasattr(socket, "TCP_KEEPCNT"):
-            # Temporarily commented out due to issues running on JupyterHub
-            # conn.sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_KEEPIDLE, TCP_KEEP_IDLE)
-            # conn.sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_KEEPINTVL, TCP_KEEPALIVE_INTERVAL)
-            # conn.sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_KEEPCNT, TCP_KEEP_CNT)
-            pass
+        if (platform == 'linux' and hasattr(conn.sock, "setsockopt") and hasattr(socket, "SO_KEEPALIVE") and
```

Comment:
Does `hasattr(conn.sock, "setsockopt")` ensure that this does not faill when setsockopt is not available?

Reply:
Correct. This is to handle the WSL use case where the socket doesn't have that function at all. I tested it yesterday, this stops it from failing entirely